Simplified/Catalog/NYPLContentTypeBadge.swift

```diff
@@ -18,19 +18,31 @@ final class NYPLContentBadgeImageView: UIImageView {
 
   @objc required init(badgeImage: NYPLBadgeImage) {
     super.init(image: UIImage(named: badgeImage.assetName()))
-    if #available(iOS 12.0, *),
-       UIScreen.main.traitCollection.userInterfaceStyle == .dark {
-      backgroundColor = NYPLConfiguration.primaryBackgroundColor
-    } else {
-      backgroundColor = NYPLConfiguration.mainColor()
-    }
+    updateColors()
     
     contentMode = .scaleAspectFit
   }
 
   required init?(coder aDecoder: NSCoder) {
     fatalError("init(coder:) has not been implemented")
   }
+  
+  override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
```

Comment:
same as above.... is this maybe to cover the case where we initialized the view in light portrait mode, then the user changed to dark mode and rotated the device?